lib/jekyll-maps/google_map_api.rb

```diff
@@ -31,7 +48,50 @@ def load_google_maps_api
             .fetch("google", {})
             .fetch("api_key", "")
           <<HTML
-<script async defer src='https://maps.googleapis.com/maps/api/js?key=#{api_key}&callback=#{Jekyll::Maps::GoogleMapTag::JS_LIB_NAME}.initializeMap'></script>
+          <script async defer>
+            
+
+            // Load maps only when DOM is loaded
+            document.addEventListener("DOMContentLoaded", function() {
+                // Maps script already loaded
+                if (window.google && window.google.maps && jekyllMaps) {
+                  jekyllMaps.initializeMap();
+                } else {
+                    enableMapsObserver();
+                }
+            });
+            
+            function enableMapsObserver() {
+              // Enable Observer on all Maps
+              var maps = document.getElementsByClassName('jekyll-map');
+
+              const observer = new IntersectionObserver(function(entries, observer) {
```

Comment:
`IntersectionObserver` is not supported in IE, right? Can we either do a feature check and only do lazy loading if `IntersectionObserver` is supported, or include a polyfill?

Reply:
Yes, that's correct. 
Sure, I'll have a look to do that and see how I can manage it 😉 

I'll let you know as soon as it is ready!

_Gerfaut_